app/services/calculate_standards.rb

```diff
@@ -1,40 +1,57 @@
-require 'dry/monads'
-class CalculateStandards
-  MODIFIERS = {1 => 1.2, 2 => 1.375, 3 => 1.725, 4 => 1.9}.freeze
-  include Dry::Monads[:result]
-
-  def initialize(id)
-    @profile = profile_find(id)
-    return nil unless @profile
+class CalculateStandards < ApplicationService
+  MALE_METABOLIC_MULTI = { common: 88.36, weight: 13.4, height: 4.8, age: 5.7 }.freeze
+  FEMALE_METABOLIC_MULTI = { common: 447.6, weight: 9.2, height: 3.1, age: 4.3 }.freeze
+  MODIFIERS = { 1 => 1.2, 2 => 1.375, 3 => 1.725, 4 => 1.9 }.freeze
+
+  attr_reader :age, :sex, :weight, :height, :activity
+  attr_accessor :coefficients
+
+  def initialize(profile)
+    @profile = profile
     @age = @profile.age
     @sex = @profile.sex
     @weight = @profile.weigh
     @height = @profile.height
-    @activity_multiplier = MODIFIERS[@profile.activity]
+    @activity = MODIFIERS[@profile.activity]
   end
 
   def call
-    return Failure(:profile_not_found) unless @profile
-    if @sex==1
-      @profile.update(BMR: mans_metabolic_rate.round)
-      Success(@profile)
-    else
-      @profile.update(BMR: womens_metabolic_rate.round)
-      Success(@profile)
-    end
+    @profile.update(BMR: calculate.round)
```

Comment:
As it's a `before_save` action I suggest to use assign attributes, not update. The hit to DB it that case will be single.